transport/http2_client.go

```diff
@@ -542,6 +534,11 @@ func (t *http2Client) handleData(f *http2.DataFrame) {
 	data := make([]byte, size)
 	copy(data, f.Data())
 	s.write(recvMsg{data: data})
+
+	if f.FrameHeader.Flags.Has(http2.FlagDataEndStream) {
+		s.readClosed()
```

Comment:
The spec does not state this explicitly, meaning the impl can make its own decision. I prefer treating this as an error because the peer is a mis-behaved server which breaks the grpc wire protocol actually.